### PR TITLE
oracled2: Get datasource from endpoint

### LINKED ISSUE #1715: Patch bandoracled2 get data source from endpoint
Currently chain store executable file as file, oracled need to query executable from endpoint

### PR DESCRIPTION
**Pull request by perimeko**
Fixed #1715

### CODE REVIEW

File: chain/cmd/bandoracled2/execute.go [hunk 1]
```diff
@@ -70,20 +72,17 @@ func SubmitReport(c *Context, l *Logger, id otypes.RequestID, reps []otypes.RawR
...
-		return nil, err
+		l.Error(":exploding_head: Failed to get data source with error: %s", err.Error())
+		return nil, nil
 	}
-	l.Debug(":balloon: Received data source #%d content: %q", id, dataSource.Executable[:32])
-	return dataSource.Executable, nil
+
+	resBytes := res.Bytes()
+
+	l.Debug(":balloon: Received data source hash: %s content: %q", hash, resBytes)
```
Comment (sorawit) [inline chain/cmd/bandoracled2/execute.go]: Please only log the first 32 bytes, similar to the previous code

File: chain/cmd/bandoracled2/execute.go [hunk 2]
```diff
@@ -70,20 +72,17 @@ func SubmitReport(c *Context, l *Logger, id otypes.RequestID, reps []otypes.RawR
...
+func GetExecutable(c *Context, l *Logger, hash string) ([]byte, error) {
+	l.Debug(":magnifying_glass_tilted_left: Fetching data source hash: %s from the endpoint", hash)
 
-	var dataSource otypes.DataSourceQuerierInfo
-	err = cdc.UnmarshalJSON(res, &dataSource)
+	res, err := grequests.Get(fmt.Sprintf(`%s/bandchain/getfile/%s`, c.chainRestServerURI, hash), nil)
 	if err != nil {
-		return nil, err
+		l.Error(":exploding_head: Failed to get data source with error: %s", err.Error())
+		return nil, nil
```
Comment (sorawit) [inline chain/cmd/bandoracled2/execute.go]: Why don't we return error here?

File: chain/cmd/bandoracled2/event.go [hunk 3]
```diff
@@ -40,9 +42,10 @@ func GetRawRequests(log sdk.ABCIMessageLog) ([]rawRequest, error) {
 		}
 
 		reqs = append(reqs, rawRequest{
-			dataSourceID: otypes.DataSourceID(dataSourceID),
-			externalID:   otypes.ExternalID(externalID),
-			calldata:     calldataList[idx],
+			dataSourceID:   otypes.DataSourceID(dataSourceID),
```
Comment (sorawit) [inline chain/cmd/bandoracled2/event.go:45]: I don't think this field is not needed anymore?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/bandoracled2/event.go (hunk 3 above)
Comment (perimeko) [reply chain/cmd/bandoracled2/event.go:45]: Still used in handler

**Review by sorawit — DISMISSED** [review]

**Comment by sorawit** [comment]
@perimeko Please fix conflict first

**Review by taobun — DISMISSED** [review]

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**